src/main/java/com/github/ermay12/regex/EvaluationType.java

```diff
@@ -0,0 +1,43 @@
+package com.github.ermay12.regex;
+
+/**
+ * This enum spcifies how to evaluate quantifiers on a regular expression.
+ * Depending on this value, quantifiers may behave differently
+ */
+public enum EvaluationType {
+    /**
+     * Evaluate the quantifier in a greedy fashion. This matches the maximum amount of characters
+     * possible while still maintaining a match.
+     *
+     * This evaluation type is the default.
+     */
+    GREEDILY,
+
+    /**
+     * Evaluate the quantifier in a possessive fashion. This matches the maximum amount of characters possible
+     * regardless of if that maintains the match.
+     * <br></br>
+     * Warning: This evaluation type may lead to failing to match a string that other quantifiers would match.
+     *          For example, <code>concatenate(anyAmount(WILDCARD, GREEDILY), string("a"))</code> will match "bba", but
+     *          <code>concatenate(anyAmount(WILDCARD, POSSESSIVELY), string("a")) will not match "bba".
+     */
+    POSSESSIVELY,
+
+    /**
+     * Evaluate the quantifier in a reluctant fashion. This matches the minimum amount of characters possible while
+     * still maintaining a match.
+     */
+    RELUCTANTLY;
+
+    String toRegex() {
+        switch(this) {
+            case GREEDILY:
```

Comment:
I think some people refer to what your calling possessive as greedy so this might be confusing.  also I think lazy is the a common term than reluctant.